src/Microsoft.IdentityModel.Tokens/SecurityKey.cs

```diff
@@ -48,13 +48,6 @@ public abstract class SecurityKey
         public string KeyId { get; set; }
 
         /// <summary>
-        /// This must be overridden to specify whether this SecurityKey supports the algorithm.
-        /// </summary>
-        /// <param name="algorithm">The crypto algorithm to use.</param>
-        /// <returns>true if this supports the algorithm; otherwise, false.</returns>
-        public abstract bool IsSupportedAlgorithm(string algorithm);
```

Comment:
@pinpointtownes
We thought it was redundant, you can call SecurityKey.CryptoProviderFactory.IsSupportedAlgorithm(algorithm, this).
This set of changes addresses a couple of issues. 
1. OpenIdConnectProtocolVerifier is disjoint from TokenValidation. You could plug in a custom algorithm but OIDCPV would still fail.
2. It was possible for a key to report YES but no CryptoSupport was actually available.
3. Plugging in a new OR overriding an existing required deriving.
4. Our delegates were scoped to AsymmetricAlgorithms and we would need one for each new type of crypto operations. In addition, the AsymmetricAlgorithm would need to be of a certain type or we would fail.